### DISCUSSION TITLE
I have a question about automatically switching custom colors between Light/Dark mode

### DISCUSSION THREAD

**Discussion by AshenOneOrz** [unocss/unocss#2288] [Q&A]
I want to define a color called primary, and I want it to be black in Light mode and white in Dark mode. How can I achieve this?

**Example:**
```html
<div color-primary-100 bg-primary-100>black</div>
```
```html
<div dark:color-primary-100 dark:bg-primary-100>white</div>
```

Thanks!

**Comment by zyyv** [comment]
Refer to [Playground](https://uno.antfu.me/play/?html=DwEwlgbgBARg5gWgA4CcwFsCGKCeCCMADIVAC4CmAHqcmlrlCNgNYLy0bZ4CcxjLCCtQ708RQgD4AUFCgB7AHYIQ5LApBQA7VDAKAzqUwLSWqADMArgBsrOKAGMLBuejAAvTDCvkpwAPTgENJAA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQ61AA9IsFBgCGbADbw0mHPkLAiACgTq4cGAAtUIVAC5ET53CUIE2gqH0d-fwpQQyhuML8I5wARAFEAMTpBABkAFR85SmQ5EgTEgEYABgr8gGIqiuLSiIBOKtr0DsbE8RKIsV6ev34aGFC4AG1hoRE7AEo%2BShGGJlYOLm45gF0SsVn1IA&css=Q&options=N4XyA)

**Reply by igorjacauna** [reply]
Is it possible to have this?:

In unocss.config.ts:

```ts
theme: {
  dark: {
    colors: {
      primary: '#FFFFFF',
    },
  },
  light: {
    colors: {
      primary: '#000000',
    },
  },
},
```

And the use will be like:

```html
<div bg-primary>Content</div>
```

So, does not need to specify `dark:{color}`, just the color. 

The generated CSS should be like:

```css
.dark .bg-primary {
  background-color: #FFFFFF;
}

.light .bg-primary {
  background-color: #000000;
}
```

This remeber the [Vuetify config for colors](https://vuetifyjs.com/en/styles/colors/#javascript-color-pack)

**Reply by igorjacauna** [reply]
I found this preset that help with this approach: https://github.com/Dunqing/unocss-preset-theme
